src/drivers/intel/cavs/idc.c

```diff
@@ -169,26 +150,30 @@ static int idc_msg_status_get(uint32_t core)
 static int idc_wait_in_blocking_mode(uint32_t target_core)
 {
 	struct timer *timer = timer_get();
-	struct idc *idc = *idc_get();
+	uint32_t core = cpu_get_id();
 	uint64_t deadline;
 
 	deadline = platform_timer_get(timer) +
 		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
 		IDC_TIMEOUT / 1000;
 
-	while (!idc->msg_processed[target_core]) {
+	while (!(idc_read(IPC_IDCIETC(target_core), core) & IPC_IDCIETC_DONE)) {
 		if (deadline < platform_timer_get(timer)) {
 			/* safe check in case we've got preempted
 			 * after read
 			 */
-			if (idc->msg_processed[target_core])
+			if (idc_read(IPC_IDCIETC(target_core), core) &
+			    IPC_IDCIETC_DONE)
```

Comment:
For future PR, we could notify the SW here that core X has crashed/timeout and do a dump